compiler/rustc_privacy/src/lib.rs

```diff
@@ -2188,7 +2193,13 @@ fn check_private_in_public(tcx: TyCtxt<'_>, (): ()) {
     }
 
     // Check for private types and traits in public interfaces.
-    let mut visitor =
-        PrivateItemsInPublicInterfacesVisitor { tcx, has_pub_restricted, old_error_set_ancestry };
+    let mut visitor = PrivateItemsInPublicInterfacesVisitor {
+        tcx,
+        has_pub_restricted,
+        old_error_set_ancestry: old_error_set_ancestry
+            .into_iter()
+            .filter_map(|hir_id| tcx.hir().opt_local_def_id(hir_id))
```

Comment:
Are there cases where we expect `tcx.hir().opt_local_def_id(hir_id)` to return `None` and it's ok to skip that `hir_id`? Or should we do `.map(|hir_id| tcx.hir().local_def_id(hir_id))` here instead?

Reply:
When the source `old_error_set_ancestry` is constructed it can indeed contain non-definition IDs, e.g. expressions.
However, only definition IDs are ever searched in the destination `old_error_set_ancestry`, so it's ok to skip non-definition IDs when converting the map.
(I should probably add a comment about this.)

Reply:
That would be great! Feel free to r=me with that comment.